src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1215,7 +1215,56 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::limit_init_tm_tx(
     }
     auto term = term_opt.value();
 
+    if (stm->tx_cache_size() > _max_transactions_per_coordinator()) {
+        // lock is sloppy and doesn't guarantee that tx_cache_size
+        // never exceeds _max_transactions_per_coordinator. init_tm_tx
+        // request may pass limit_init_tm_tx but not yet increase
+        // tx_cache_size so there is a small window of time when the
+        // next init tx request may pass too even if the first request
```

Comment:
can we roll this log into the expire_timer? (make it run more frequently but consider both expiration and count based evictions in one place).

Reply:
it feels like a bigger change, it requires figuring out how often we should trigger it and reviewing if cranking the frequency doesn't have perf implications: at the moment the expiration is low freq event since it isn't about (auto-)aborting an open transaction (by default 10s) but about garbage collecting old sessions (by default 7 days).

We have a [todo](https://github.com/redpanda-data/redpanda/issues/9606) item to modify the expiration logic so I linked this comment if we decide to do it later.